File: Authentication/Model/RequestModels/RecipeModel.cs


using Microsoft.AspNetCore.Http;

namespace Model.RequestModels
{
    public class RecipeModel
    {
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string Category { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;

     //   public List<IFormFile> ImageData { get; set; }  

        public List<string> Ingredients { get; set; } = new();
        public List<string> Instructions { get; set; } = new();
    }
}

File: Authentication/AuthAPI/ClientAPP/src/app/Models/Intefaces/Recipe.model.ts
import { Ingredient } from "./Ingredient.model";
import { Instruction } from "./Instruction.model";
import { ApplicationUser } from "./ApplicationUser.model";
import { Rating } from "./Rating.model";

export interface Recipe {
  id: number;
  title: string;
  description: string;
  category: string;
  imageUrl: string;

  userId: string;
  user?: ApplicationUser;

  instructions: Instruction[];
  ingredients: Ingredient[];
  comments: Comment[];
  ratings: Rating[];
}
